roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_ocp_virt/configure-worker-node.yml

```diff
@@ -14,6 +9,22 @@
 
 # How to wait for node to be scheduleable? (NodeSchedulable)
 - name: Wait for all k8s nodes to be ready
-  ansible.builtin.command: oc wait --for=condition=Ready nodes --all --timeout=3600s
-  register: __sap_hypervisor_node_preconfigure_register_nodes_ready
-  changed_when: __sap_hypervisor_node_preconfigure_register_nodes_ready.rc != 0
+  kubernetes.core.k8s_info:
+    host: "{{ sap_hypervisor_node_preconfigure_ocp_endpoint }}"
+    ca_cert: "{{ sap_hypervisor_node_preconfigure_ocp_ca_cert }}"
+    api_key: "{{ __sap_hypervisor_node_preconfigure_register_openshift_auth_results.openshift_auth.api_key }}"
+    kubeconfig: "{{ sap_hypervisor_node_preconfigure_ocp_kubeconfig }}"
+    kind: Node
+  register: __sap_hypervisor_node_preconfigure_register_nodes
+  until: __sap_hypervisor_node_preconfigure_register_nodes.resources | json_query('[?status.conditions[?type==`Ready` && status==`True`]].metadata.name') |\
+         length == __sap_hypervisor_node_preconfigure_register_nodes.resources | length
+  retries: 90
+  delay: 20
+  ignore_errors: true
+  changed_when: false
+
+- name: Check and fail if any node is not ready
```

Comment:
Could this be avoided by adding a `failed_when` above?

Reply:
Unfortunately, `k8s_info` does not provide `failed_when`.

Reply:
Isn't that a general Ansible task option?